src/core_modules/capture-core/components/Pages/New/NewPage.component.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import React, { useEffect } from 'react';
+import { compose } from 'redux';
+import type { ComponentType } from 'react';
+import withStyles from '@material-ui/core/styles/withStyles';
+import { LockedSelector } from '../../LockedSelector';
+import type { ContainerProps, Props } from './NewPage.types';
+import { withErrorMessageHandler, withLoadingIndicator } from '../../../HOC';
+import { newPageStatuses } from './NewPage.constants';
+
+const getStyles = () => ({
+    container: {
+        padding: '10px 24px 24px 24px',
+    },
+});
+
+const NewPagePlain = ({
+    showMessageToSelectOrgUnitOnNewPage,
+    showDefaultViewOnNewPage,
+    classes,
+    currentOrgUnitId,
+    newPageStatus,
+}: Props) => {
+    useEffect(() => {
+        if (!currentOrgUnitId) {
```

Comment:
I know we discussed these statuses before, but this is good example. Here you could just check if there is currentOrgUnitId directly when the page is rendered instead of dispatching an action. If you want to, you could compute the status in this component instead of saving it in the Redux store.

Reply:
Right. Perhaps we could look at an example where this pattern adds some value.

This technique shines and works very well with epics and asynchronous actions.
An example would be this: 

```
const searchViaAttributesStream = (queryArgs, attributes, triggeredFrom) => {
    const stream: Stream = from(getTrackedEntityInstances(queryArgs, attributes));
    return stream.pipe(
        map(({ trackedEntityInstanceContainers: searchResults, pagingData }) => {
            if (searchResults.length > 0) {
                return showSuccessResultsViewOnSearchPage(
                    searchResults,
                    pagingData.currentPage,
                );
            }
            return showEmptyResultsViewOnSearchPage();
        }),
        startWith(showLoadingViewOnSearchPage()),
        catchError(() => of(showErrorViewOnSearchPage())),
    );
};
```

Where I want the page to be in loading state when the user is triggering the request and I want to show an error when errors are being shown. 

I am gonna try to explain the biggest advantage of having this string, instead of `boolean` values or computed values. 

Basically when you have boolean values for `loading` and `error` you can be on a state that both `loading` and `error` can be true. Whereas in actuality there is no point for your page to be in that state. 

State machines is common pattern in computer science in general. Every arrow in the image is a user action and can take them to one state. 
![image](https://user-images.githubusercontent.com/4181674/98092388-e4529380-1e86-11eb-9498-45b4e5825e10.png)

Here are some other resources 
[1]https://www.youtube.com/watch?v=VU1NKX6Qkxc


What are some drawbacks you see?

Reply:
Thinking about this a bit, let's go with the solution you proposed 👍 I think the defined statuses earlier was what threw me off a bit.